File: logic/estimate_core.php
<?php
declare(strict_types=1);

/**
 * Estimate core calculation logic (PHP)
 * Mirrors JS/Python core to validate business rules independent of UI.
 */

final class EstimateParams {
    public float $area;
    public float $unitPrice;

    public function __construct(float $area, float $unitPrice) {
        $this->area = $area;
        $this->unitPrice = $unitPrice;
    }
}

function calculate_estimate(EstimateParams $params): float {
    if ($params->area <= 0 || $params->unitPrice <= 0) {
        throw new InvalidArgumentException('Values must be greater than zero');
    }
    return $params->area * $params->unitPrice;
}

File: logic/estimate_core.py
"""
Estimate core calculation logic (Python)
This file mirrors the JS core to validate business rules independent of UI.
"""

from dataclasses import dataclass


@dataclass(frozen=True)
class EstimateParams:
    area: float
    unit_price: float


def calculate_estimate(params: EstimateParams) -> float:
    if params.area is None or params.unit_price is None:
        raise ValueError("Invalid input for estimation")

    if params.area <= 0 or params.unit_price <= 0:
        raise ValueError("Values must be greater than zero")

    return params.area * params.unit_price
